### PR TITLE
AUI-2820

### PR DESCRIPTION
**Pull request by jonmak08**
Hey @thiago-rocha,

Attached is an update for https://issues.liferay.com/browse/AUI-2820.

Please let me know if you have any questions. Thanks!

### CODE REVIEW

**Comment by thiago-rocha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by thiago-rocha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: src/aui-form-validator/js/aui-form-validator.js [hunk 1]
```diff
@@ -734,12 +734,12 @@ var FormValidator = A.Component.create({
...
          * @method getFieldStackErrorContainer
-         * @param {Node} field
+         * @param {Node|String} field
          * @return {Node}
          */
         getFieldStackErrorContainer: function(field) {
             var instance = this,
-                name = field.get('name'),
+                name = isNode(field) ? field.get('name') : field,
                 stackContainers = instance._stackErrorContainers;
```
Comment (thiago-rocha) [inline src/aui-form-validator/js/aui-form-validator.js]: @blzaugg what you think about apply this improve on the `highlight` function?

Comment (blzaugg) [inline src/aui-form-validator/js/aui-form-validator.js]: I've thought about doing this across all uses of `field`, to always return the "correct" one. But declined to do it here, for this ticket.

If we want to generally improve the code to support that, I figured we could do it on a dedicated ticket.

File: src/aui-form-validator/tests/unit/js/tests.js [hunk 2]
```diff
@@ -602,6 +602,160 @@ YUI.add('aui-form-validator-tests', function(Y) {
             Y.Assert.isFalse(input.hasClass('success-field'));
         },
 
+        /*
+         * @tests AUI-2820
+         */
+        'should add success-field and aria-invalid attr on all, named Nodes and their fieldContainer': function() {
```
Comment (thiago-rocha) [inline src/aui-form-validator/tests/unit/js/tests.js]: This 2 tests are failing on my machine when run with gulp test, could you run `gulp test`  check this?

✖ [aui-form-validator]: Passed: 17 Failed: 2 Total: 19 (ignored 0) (0.254 seconds)
    should add success-field class to, and remove aria-invalid attr from all, named Nodes and their fieldContainer
       All fields should have success-field
       Expected: true (boolean)
       Actual: false (boolean)
    should add success-field class to, and remove error-field class and aria-invalid attr from all, named Nodes and their fieldContainer
       All fields should have success-field
       Expected: true (boolean)
       Actual: false (boolean)

**Comment by thiago-rocha** [comment]
Pull request submitted to https://github.com/mairatma/alloy-ui/pull/340. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**